### PR TITLE
Billable resource report changes

### PR DESCRIPTION
**Pull request by tejaszambre**
+ Added Designation field in billable report(first sheet)
+ Splited project column in multiple columns with one project name in each column in billable report(first sheet)
+ Splitted the technical skill column in three columns, almost three skill are displayed in billable resource report
+ Added UI/UX ladder and devops engg at end of the billable resource report with non-billable allocation 160
+ Added all the missing entries at the end of the billable resource report with bench allocation 160
+ Filtered all names where allocation is zero in project wise resource report
+ Added allocation hrs under Non-billable allocation for employees who are marked as Non-billable for that project.
+ Added 3 designations to exclude designations
+ Limit technical skills to max 3
+ Added validation on technical skills in public profile to have maximum 3 length
+ Change caption of Technical Skills to Core Technical Skills
+ Updated values of Language, Framework and Other global constants as per requirements
+ Added rake task move other skills which moves the skills other than skills in core technical skill set
+ Added rake task max three technical skills to keep only 3 skills in technical skills field and move all other skills in skills field
+ Fixed the technical skill update blank issue
+ Added constant to list UI and UX designation
+ Added method to get project names whose allocation hrs is greater than 0
+ Added Rake Task For Mapping Current Skill Set To New Skill Set
+ Added shared column in the RC report
+ Shared column will be filled only for shared resources which are:
	1. UI/UX ladder
	2. DevOps engineer
	3. QA Lead
	4. Team Lead
	5. Technical Manager
	6. Senior Technical Manager
	7. Project Manager
	8. Delivery Manager
	9. Delivery Head
+ Added Manager Role in the Report
+ Added shared resources(as mentioned above) at bottom of the report and before bench resources
+ Corrected the calculation for total allocation
+ Added condition for non_billable, investment and shared allocation to not exceed 160 allocation
+ Modified the testcases for above

### CODE REVIEW

**Comment by tejaszambre** [comment]
[Billable resource report changes](https://trello.com/c/2Y6Nwb2O/15-billable-resource-report-changes)

**Comment by tejaszambre** [comment]
[Changes in resource categorisation report](https://trello.com/c/1V0iuxJz/3-changes-in-resource-categorisation-report)

File: app/services/resource_categorisation_service.rb [hunk 1]
```diff
@@ -11,10 +11,24 @@ def call
   end
 
   def generate_resource_report
+    bench_resource = []
+    devops_ui_ux_resource = []
+    ui_ux_designation = [
```
Comment (Sameer-Tilak) [inline app/services/resource_categorisation_service.rb:16]: Shift this ui_ux_designation array to constant

File: app/services/resource_categorisation_service.rb [hunk 2]
```diff
@@ -27,21 +41,42 @@ def generate_resource_report
         bench_allocation =  (160 - total_allocation) < 0 ? 0 : (160 - total_allocation)
         project_names = user.project_details.map { |i| i.values[1] }
```
Comment (Sameer-Tilak) [inline app/services/resource_categorisation_service.rb:42]: Collect only those projects of any employee where hrs allocation is > 0. Filter out all projects from list with hrs allocation = 0.

File: config/initializers/global_constants.rb [hunk 3]
```diff
@@ -5,9 +5,9 @@
 LEAVE_STATUS = ['Pending', 'Approved', 'Rejected']
 INVALID_REDIRECTIONS = ["/users/sign_in", "/users/sign_up", "/users/password"]
 TSHIRT_SIZE = ['XS', 'S', 'M', 'L', 'XL', 'XXL', 'XXXL']
-OTHER = [ 'Delivery Management', 'Design', 'DevOps', 'QA-Automation', 'QA-Manual', 'UI', 'UX']
-LANGUAGE = ['Go', 'Python', 'Ruby', 'Java', 'Javascript', 'PHP', 'Android', 'NodeJs', 'React', 'iOS', 'C#']
-FRAMEWORK = ['Django', 'Rails', 'Spring Boot', 'Hibernet', 'Laravel', 'Angular', 'Flutter', 'Ionic', '.Net']
+OTHER = ['DevOps', 'QA-Automation', 'QA-Manual', 'UI', 'UX']
+LANGUAGE = ['Go', 'Python', 'Ruby', 'Java', 'PHP', 'Android', 'NodeJs', 'iOS']
+FRAMEWORK = ['ReactJs', 'Angular', 'Flutter', '.Net']
```
Comment (Sameer-Tilak) [inline config/initializers/global_constants.rb:10]: Shift NodeJs in FRAMEWORK array from LANGUAGE array.

File: lib/tasks/update_public_profiles.rake [hunk 4]
```diff
@@ -1,73 +1,105 @@
-desc 'Update the Github and Twitter handle of all users'
-task :update_public_profile => :environment do
+namespace :update_data do
+  desc 'Move technical skills other than core technical skill set to skills of all employees'
+  task move_other_skills: :environment do
+    TECHNICAL_SKILLS_SET = LANGUAGE + FRAMEWORK + OTHER
```
Comment (Sameer-Tilak) [inline lib/tasks/update_public_profiles.rake:4]: Where is mapping of existing technical skills and new core skills.
As Existing Rails, Ruby should be replaced by Ruby
Existing Laravel, Django should be replaced by Python.
Hibernet, Springboot should be replaced by Java.

And original values need to be copied to other skills column.

Please share this mapping as you understand between current skill set and new proposed skill set with me.

Comment (tejaszambre) [reply lib/tasks/update_public_profiles.rake:4]: New skills set(new proposed)     <-    Current skill set

NodeJs OR ReactJs OR Angular <- Javascript
Angular <- Ionic
ReactJs <- React
Ruby <- Rails, ROR
Python <- Django
PHP <- Laravel
Java <- Hibernate, Spring Boot

Not sure about below 

UX OR UI <- Design
Devops <- Delivery Management

**Comment by Sameer-Tilak** [comment]
LGTM. Take data backup before running script to change Core skill info. Once script is executed do random checking for correctness, in case of any issue rollback data by restoring db backup.

**Review by Ashish944 — APPROVED** [review]

**Pull request merged**